### ISSUE TITLE
Unable to reproduce experimental results

### ISSUE THREAD

**Issue by awslsky** [zzzhhzzz/Ground-R1#3]
Hi, we have tried to reproduce the experimental results based on your open-sourced code and the settings described in the paper, such as setting G1=4 and not using KL divergence. However, we encountered some issues during the reproduction process.

We noticed that the training curve behaves abnormally — the refine count drops to 1 very quickly, and the response length reduces to around 20 tokens. The model often generates outputs like "\<think\>Reason\<\\think\>\<answer\>xxx\<\\answer>", with no actual reasoning process included. If possible, we would appreciate it if you could provide more comprehensive parameter settings (e.g., training_args) to help us better reproduce the results.

<img width="698" height="500" alt="Image" src="https://github.com/user-attachments/assets/a0a9dd1a-c4a4-4659-9cc2-ff8bfa6a054a" />

<img width="686" height="500" alt="Image" src="https://github.com/user-attachments/assets/e8fdb6ff-62c4-4e39-98c4-fc10da425abf" />

<img width="694" height="494" alt="Image" src="https://github.com/user-attachments/assets/2ec4c485-4bc6-4227-b97c-bec05c22d82a" />

Additionally, we are currently testing the code on the LVLM benchmark datasets and have observed some discrepancies. For example, the MME score is only 2163.1, which is significantly lower than the value reported in the paper. When evaluating on MM-Vet, the reported accuracy is nearly zero, despite the fact that we visually confirmed that some of the model outputs are correct. We suspect that there might be an issue with the evaluation script.

We would greatly appreciate your help in addressing these concerns.

**Comment by zzzhhzzz** [comment]
Hi, thank you for your interest in our work and for the detailed feedback.

Completely removing the KL divergence term can indeed lead to unstable training and degenerate reasoning patterns (e.g., outputs like "<think>Reason<\think><answer>xxx<\answer>"). To improve stability and encourage meaningful reasoning steps, we recommend setting a small KL coefficient — for example, using `--beta 0.04` during training.

For better performance reproduction, you may also consider using `train_data/train_deepeyes.jsonl` as the training data. This setting may lead to even higher scores on certain benchmarks than those reported in the paper.

As for the MM-Vet results: Evaluation should be conducted using the official online evaluator provided by the [MMVet](https://huggingface.co/spaces/whyu/MM-Vet_Evaluator).

**Comment by DwanZhang-AI** [comment]
Hi, I noticed that your paper mentions using 8k data for RL training, but the accompanying script appears to use 33k. Could you clarify which setting corresponds to the results reported in the paper?

**Comment by DwanZhang-AI** [comment]
Also, you mentioned that using KL coefficient is necessary during training, but the paper demonstrates that removing KL constraints yields superior performance in the Ground-R1. How does it happen?

**Comment by zzzhhzzz** [comment]
> Hi, I noticed that your paper mentions using 8k data for RL training, but the accompanying script appears to use 33k. Could you clarify which setting corresponds to the results reported in the paper?

Hi, we only trained for 1000 steps in the RL phase, so the total number of samples used was 8k, which matches what's reported in the paper.

**Comment by zzzhhzzz** [comment]
> Also, you mentioned that using KL coefficient is necessary during training, but the paper demonstrates that removing KL constraints yields superior performance in the Ground-R1. How does it happen?

While the original Ground-R1 used 8K data with no KL constraint, we found that when expanding the dataset diversity (to 33k, but still 8k samples was used), adding KL regularization helps stabilize training

**Comment by An-LanWang** [comment]
@DwanZhang-AI Hello, have you managed to reproduce the experimental results? I'm encountering the same issue. Despite following the instructions meticulously—training the model for 1000 steps and incorporating KL regularization—there remains a noticeable performance gap between my results and those reported in the paper.

**Comment by Pefect96** [comment]
Yes. I met the same questions, and I do not remove the KL divergence. After 1000 steps, the MME score is only 2191.1, @DwanZhang-AI @awslsky @zzzhhzzz @An-LanWang Have you solved this problem?
